### PR TITLE
[doc] Add Root Keys documentation.

### PR DESCRIPTION
**Pull request by moidx**
Add documentation associated with the "Move ROM keys to OTP" RFC:  #21204. Updates sigverify module documentation and removes the now outdated rom/keys/README.md.

### CODE REVIEW

File: sw/device/silicon_creator/rom/doc/root_keys.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+manufacturing time, similar to the Darjeeling architecture. This allows the
+Silicon Creator to assign keys based on SKU configuration, enabling key rotation
+and also provisioning of keys for Silicon Owners that act as effective Silicon
+Creators.
+
+## Supported Keys
+
+OpenTitan is required to support hybrid signature schemes for secure boot. For
+Earl Grey, this is based on ECDSA-P256-SHA256 and SLH-DSA. In this document and
+in the code base we use Sphincs+ and spx to refer to SLH-DSA.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/doc/root_keys.md:16]:
```suggestion
in the code base we use "SPHINCS+", "spx", and/or "spx+" to refer to SLH-DSA.
```
?

File: sw/device/silicon_creator/rom/doc/root_keys.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+All keys in the current Earl Grey implementation use a common `key_type` and a
+`key_id` field. The `key_id` represents the first `uint32_t` word of the public
+key, independent of whether the key is ECDSA-P256 or SLH-DSA.
+
+The `key_type` determines the life cycle states in which the key is enabled for
+secure boot verification. The value can be one of the following:
+
+*   `kSigverifyKeyTypeInvalid`: Applicable to empty key slots.
+*   `kSigverifyKeyTypeTest`: Applicable to devices in TEST\_UNLOCKED and RMA
+    state.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/doc/root_keys.md:32]:
```suggestion
    states.
```

File: sw/device/silicon_creator/rom/doc/root_keys.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+*   `kSigverifyKeyTypeProd`: Applicable to devices in PROD, PROD\_END or DEV
+    state.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/doc/root_keys.md:34]:
```suggestion
    states.
```

File: sw/device/silicon_creator/rom/doc/root_keys.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    *   In DEV state, debug unlock functionality is allowed after the ROM\_EXT
+        hands over execution to the first Silicon Owner code partition.
+*   `kSigverifyKeyTypeDev`: Applicable of devices in DEV state and signed with a
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/doc/root_keys.md:37]:
```suggestion
*   `kSigverifyKeyTypeDev`: Applicable to devices in DEV state and signed with a
```

File: sw/device/silicon_creator/rom/doc/root_keys.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+The keys are provisioned at manufacturing time when the device is in one of the
+`TEST_UNLOCKED` states. `KEY0` will be provisioned with a key for
+`TEST_UNLOCKED` and `RMA` life cycle states. `KEY1` to `KEY3` are provisioned
+with mission mode keys (i.e. `kSigverifyKeyTypeProd` or `kSigverifyKeyTypeDev`).
+Mission mode keys are managed by the SKU owner, which may be an entity
+representing the Silicon Creator or the Silicon Owner.
+
+There must be at least one valid key provisioned in the device before enabling
+ROM execution.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/doc/root_keys.md:107]: nit: maybe **bold** this?

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/rom/doc/root_keys.md (hunk 5 above)
Comment (moidx) [reply sw/device/silicon_creator/rom/doc/root_keys.md]: Fixed.

**Pull request merged**